src/Context.Tests/ImmutableConventionTests.cs

```diff
@@ -122,6 +122,23 @@ public async Task ApplyConvention_WithoutValueInDb_SerializeSuccessful()
                 result.MatchSnapshot();
             }
 
+            [Fact]
+            public async Task ApplyConvention_WithoutValueInDbWithoutDefault_SerializeSuccessful()
```

Comment:
I would create an additional test case, in which you InsertOneAsync with the C, like in ApplyConvention_WithoutValue_SerializeSuccessful, just to make sure in future, that the serializing works as well.

Reply:
Just a proposal

Reply:
I created `ApplyConvention_CtorWithDefault_SerializeSuccessful` to cover this case